### PR TITLE
HPO code flow modification

### PR DESCRIPTION
**Pull request by eunwoosh**
This PR modifies HPO code flow as below

- Remove unnecessary HPO algo class attribute
- Exclude anomaly Padim model from HPO because the model doesn't need to be trained.
- Add metric_name to HpoConfig. User can configure which metric to use for HPO
- Enable HPO api test
- If metric is loss, then change hpo mode to 'min'
- HPO raises an error if task is Zeroshot VP or model is padim
- Update integration test and e2e test
- Fix a bug that getting iteration per epoch instead of length of dataset

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by vinnamkim — CHANGES_REQUESTED** [review]
Please revisit this after merging https://github.com/openvinotoolkit/training_extensions/pull/3258

File: src/otx/engine/hpo/hpo_api.py [hunk 1]
```diff
@@ -60,8 +60,13 @@ def execute_hpo(
...
             return None.
     """
+    skip_msg = ""
     if engine.task == OTXTaskType.ZERO_SHOT_VISUAL_PROMPTING:  # type: ignore[has-type]
-        logger.warning("Zero shot visual prompting task doesn't support HPO.")
+        skip_msg = "Zero shot visual prompting task doesn't support HPO."
+    elif "anomaly.padim" in str(type(engine.model)).lower():
+        skip_msg = "Padim doesn't need HPO. HPO is skipped."
+    if skip_msg:
+        logger.warning(skip_msg)
```
Comment (vinnamkim) [inline src/otx/engine/hpo/hpo_api.py]:
```suggestion
        msg = "Zero shot visual prompting task doesn't support HPO."
        raise RuntimeError(msg)
    elif "anomaly.padim" in str(type(engine.model)).lower():
        msg = "Padim doesn't need HPO. HPO is skipped."
        raise RuntimeError(msg)
```

**Review by jaegukhyun — DISMISSED** [review]

File: src/otx/engine/hpo/hpo_api.py (hunk 1 above)
Comment (eunwoosh) [reply src/otx/engine/hpo/hpo_api.py]: fixed https://github.com/openvinotoolkit/training_extensions/pull/3259/commits/b834d0844fc2b55f6c271d543a598f4ca24d49a3

**Review by vinnamkim — DISMISSED** [review]

**Review by jaegukhyun — DISMISSED** [review]

**Review by jaegukhyun — DISMISSED** [review]

File: src/otx/algo/classification/efficientnet_b0.py [hunk 2]
```diff
@@ -101,6 +101,9 @@ def load_from_otx_v1_ckpt(self, state_dict: dict, add_prefix: str = "model.model
         """Load the previous OTX ckpt according to OTX2.0."""
         return OTXv1Helper.load_cls_effnet_b0_ckpt(state_dict, "multiclass", add_prefix)
 
+    def _reset_prediction_layer(self, num_classes: int) -> None:
+        return
```
Comment (harimkang) [inline src/otx/algo/classification/efficientnet_b0.py:105]: why we need this change?

**Review by vinnamkim — APPROVED** [review]

File: src/otx/algo/classification/efficientnet_b0.py (hunk 2 above)
Comment (eunwoosh) [reply src/otx/algo/classification/efficientnet_b0.py:105]: If it doesn't exist, training this model doesn't work

**Review by jaegukhyun — APPROVED** [review]
